### PR TITLE
US22125 Reduce Log Verbosity and add page limit

### PR DESCRIPTION
**Pull request by mjvezzani**
This PR implements the following:

- Reduce Log Verbosity
- Add a --pagelimit flag

The new --pagelimit flag gives control to the jekyll contentful command
to dictate how many results come back with each page. For example, if
Artlce has 100 total records in Contentful, using the --pagelimit flag
with a value of 30 would result in 4 API calls that returned a total of
100 records.

This differs from the --limit flag, in which you specify the total
number of records you want returned. For example, --limit 10 would
result in 1 API call that returned 10 records.

### CODE REVIEW

**Review by silasbaur — APPROVED** [review]
Approved! We may want to spend some time creating a GitHub Action instead of Travis CI.

File: lib/jekyll-contentful/client.rb [hunk 1]
```diff
@@ -98,10 +99,11 @@ def space
       private
 
         def fetch_entries(type)
+          page_limit = options.dig('pagelimit')
           @entries[type] ||= []
           params = {
             skip: @entries[type].count,
-            limit: (options.dig('limit') || @limit)
+            limit: (page_limit || options.dig('limit') || @limit)
```
Comment (tcmacdonald) [inline lib/jekyll-contentful/client.rb:106]: Can you please consolidate all of this logic within the constructor so we can simplify this method? Something like this... 

```
@limit = options.dig('pagelimit') || options.dig('limit') || 1000
```

File: lib/jekyll-contentful/client.rb [hunk 2]
```diff
@@ -113,13 +115,12 @@ def fetch_entries(type)
...
-          log("Querying '#{type.pluralize}' with the following parameters...")
-          log(params.to_json)
-
           this_page = type == 'asset' ? client.assets(params).to_a : client.entries(params).to_a
           @entries[type].concat(this_page)
 
-          if this_page.size == @limit
+          if page_limit && this_page.size == page_limit
+            fetch_entries(type)
+          elsif this_page.size == @limit
```
Comment (tcmacdonald) [inline lib/jekyll-contentful/client.rb:123]: See my comment above ☝🏻 regarding consolidation/simplification.

**Review by tcmacdonald — CHANGES_REQUESTED** [review]
@mjvezzani Sorry for the delay. Left one request, let me know if you have questions.

File: lib/jekyll-contentful/client.rb (hunk 1 above)
Comment (mjvezzani) [reply lib/jekyll-contentful/client.rb:106]: So I see one issue with doing this. Right now `@limit` in the constructor gives us the threshold that we use to determine if we need to make follow-up API calls. When the `--limit` flag gets passed in at the command line, usually that limit is less than the `@limit` in the constructor. This is useful if a developer wants to pull down a smaller content payload, resulting in quicker dev cycles.

If we make the `@limit` in the constructor equal the `--limit` flag, then we will alway end up making subsequent API calls until we've pulled all the content from contentful. On line 123 you can see that we check to see if the `this_page.size == @limit`. Assuming a `@limit` of 1000, if the `--limit` flag value is 50, then the block of code to call `fetch_entries` again doesn't execute, resulting in just 50 content records getting returned.

This is further complicated by the new `--pagelimit` flag. If `--pagelimit` is present, then we want to make sure we call the API multiple times until there are no more records to return, hence the extra `if` clause on lines 121-22. If `page_limit` exists and the `this_page.size` equals our `page_limit`, then we want to make additional calls.

TL;DR - If we convolute `@limit` with `--limit` and `--pagelimit`, then we will always end up calling the API until all records get returned. I'm going to look at how some of these variables might be renamed so as to clear up confusion, but functionally speaking, there is a difference between `--limit` and `--pagelimit` such that we can't indiscriminately assign either value to the `@limit` constructor variable.
